plugins/destination/batchingwriter/options.go

```diff
@@ -0,0 +1,34 @@
+package batchingwriter
+
+import (
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+type Option func(*Batching)
+
+func WithLogger(l zerolog.Logger) Option {
+	return func(w *Batching) {
+		w.logger = l
+	}
+}
+
+func WithDedupPK(dedup bool) Option {
```

Comment:
I'd make it `WithNoPKDedup()` so that it'll be obvious that the default is to dedup (or removed altogether)

Reply:
In that case there would be no explicit way of specifying what you want though, depending on the default behavior set by that specific SDK version? (As `WithNoPKDedup(bool)` would be funny)